lib/srv/regular/sshserver.go

```diff
@@ -359,6 +368,11 @@ func (s *Server) Start() error {
 			return trace.Wrap(err)
 		}
 	}
+	// clean up sudoers before starting, in case any were left behind
+	// due to a crash for example
+	if err := s.sudoers.CleanupSudoers(); err != nil {
+		return trace.Wrap(err)
+	}
```

Comment:
@espadolini can correct me if I'm wrong but I suspect that this might cause us to nuke sudoers for existing connections during graceful restart.

I don't know if we should be doing this sort of cleanup when starting the server. Can we just clean up sudoers for a particular user before their respective connection (unless they're already connected)?

Reply:
Yes, we will reach this point long before old sessions are guaranteed to be finished.

Do we _have_ to delete old sudoers files? I would very much prefer to be conservative about deleting system-level things.

Reply:
I think it should be fine to not do the sudoers cleanup at startup time. The current behavior is the clean up the sudoers at session end so i think it makes sense to keep that.

We shouldnt need to remove the sudoers file before connection starts, it gets overwritten anyway with whatever the contents of the current host sudoers permission is